### PR TITLE
[BugFix] Fix BE schema scan node data distribution

### LINKED ISSUE #23306: Aggregate on BE schema tables not correct
### Real behavior (Required)
Seems aggregation or data distribution spec is not right for BE schema tables.
```
mysql> select table_id, tablet_id, count(*) as cnt from be_tablets where tablet_id = 148811 group by table_id, tablet_id;
+----------+-----------+------+
| table_id | tablet_id | cnt  |
+----------+-----------+------+
|   142841 |    148811 |    1 |
|   142841 |    148811 |    1 |
|   142841 |    148811 |    1 |
+----------+-----------+------+
3 rows in set (0.24 sec)
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by decster**
## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes:

Fixes #23306

## Problem Summary(Required):

This PR fixes aggregation on BE schema tables.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3

### CODE REVIEW

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/InformationSchemaBeFeTableTest.java [hunk 1]
```diff
@@ -62,6 +62,30 @@ public void testQueryBeSchemaTables() throws Exception {
...
+        Statement stmt = connection.createStatement();
+        try {
+            // https://github.com/StarRocks/starrocks/issues/23306
+            // verify aggregation behavior is correct
+            Assert.assertTrue(stmt.execute(
+                    "explain select table_id, count(*) as cnt from information_schema.be_tablets group by table_id"));
+            int numExchange = 0;
+            while (stmt.getResultSet().next()) {
+                String line = stmt.getResultSet().getString(1);
+                System.out.println(line);
```
Comment (chaoyli) [inline fe/fe-core/src/test/java/com/starrocks/sql/plan/InformationSchemaBeFeTableTest.java:77]: This output is not necessary

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/OutputPropertyDeriver.java|3|3|100.00%|[]|

**Review by Seaven — DISMISSED** [review]

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/InformationSchemaBeFeTableTest.java (hunk 1 above)
Comment (decster) [reply fe/fe-core/src/test/java/com/starrocks/sql/plan/InformationSchemaBeFeTableTest.java]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/OutputPropertyDeriver.java|3|3|100.00%|[]|

**Review by chaoyli — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.3

**Pull request merged**